app/views/components/links/ChoicePageBackLink.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.components.links
+import models.UcrBlock
+import play.api.mvc.Call
+
+object ChoicePageBackLink {
+  def call(ileQueryEnabled: Boolean, queryUcr: Option[UcrBlock]): Call =
```

Comment:
This flag could be injected and this object could be moved as injeted class to view - this would does not require to pass flags and appConfig to view